### DISCUSSION TITLE
Index Operator Interface

### DISCUSSION THREAD

**Discussion by ggpwnkthx** [commonism/aiopenapi3#164] [Ideas]
An `OperationIndex` that has an ID containing a `.` (e.g. `me.GetProfile`) can't be accessed through the sad smiley interface without adding some kind of chained attribute functionality - at least to stay aligned with the spirit of accessors. I wrote something up for that, but it's feels hacky and I don't like it.

Instead, how do you feel about an index operator for the `OpenAPI` class?
``` python
class OpenAPI:
  ...
    def __getitem__(self, item: str) -> OperationIndex:
        """
        index operator interface
        access operations by operationId
        """
        return getattr(self._, item)
```
``` python
api = OpenAPI.loads(...)
api["me.GetProfile"].request()
```

This could be expanded on to handle the `api[path, method]` syntax pretty easily, too.
``` python
class OpenAPI:
  ...
    def __getitem__(self, item: Union[str, Tuple]) -> OperationIndex:
        """
        index operator interface
        access operations by operationId
        """
        return self.createRequest(item) if isinstance(item, tuple) else getattr(self._, item)
```

**Comment by commonism** [comment]
I'd prefer to have this on OperationIndex:
```python
api._["me.GetProfile"]
api._[(path, method)]
```

As I see this - \_\_call\_\_ can be used instead of request()

```python
api._["me.GetProfile"](parameters=dict(user="…"))
```

**Reply by ggpwnkthx** [reply]
Sounds good. I like the `__call__` idea as well!
I've wrapped up work for today but I can submit a PR tomorrow.

**Reply by commonism** [reply]
Request.__call__ already exists, I mentioned this as I saw you do not use it

https://github.com/commonism/aiopenapi3/blob/55b200da7bb3c8b39ee2e71c5a15a25acef81962/aiopenapi3/request.py#L96-L106

it's used implicit when using the ssi
```
api._.getUsers(parameters=dict(groups=["…"]))
```

**Reply by ggpwnkthx** [reply]
I see (said the blind man... lol). I can't believe I missed that. Thank you.
